packages/amplify-graphql-api-construct/src/types.ts

```diff
@@ -496,6 +496,13 @@ export interface TranslationBehavior {
    * @experimental
    */
   readonly replaceTableUponGsiUpdate: boolean;
+
+  /**
+   * When disabled usage of Gen 1 patterns will result in an error thrown.
+   *
```

Comment:
Are we planning to describe what these Gen-1 patterns are in a documentation update? If not, I would just call them out here. If a complete list is more verbose, we can include couple of examples to give a better idea in terms of where it applies.

Reply:
This isn't fully intended to be used by customers. This will be set to false in `defineData` and won't be configurable by customers. Customers could set this when using the construct directly, but there isn't really a use case for this because it will only restrict the features available in the construct.

Reply:
1. Agree with the intent, but I notice that we don't flag this as an internal API. Please add a doc comment noting it so customers understand that we reserve the right to change this API without a breaking change
2. For our own future selves, can you add one or two examples, like "e.g., @manyToMany, fields-style references, @predictions, etc", so we know what "Gen 1 patterns" means in 6 months? :)